src/app/CASESessionManager.cpp

```diff
@@ -52,51 +52,60 @@ void CASESessionManager::FindOrEstablishSession(const ScopedNodeId & peerId, Cal
     }
 
     session->Connect(onConnection, onFailure);
-    if (!session->IsConnected() && !session->IsConnecting() && !session->IsResolvingAddress())
-    {
-        // This session is not making progress toward anything.  It will have
-        // notified the consumer about the failure already via the provided
-        // callbacks, if any.
-        //
-        // Release the peer rather than the pointer in case the failure handler
-        // has already released the session.
-        ReleaseSession(peerId);
-    }
 }
 
+// TODO need to figure out what things are calling this
 void CASESessionManager::ReleaseSession(const ScopedNodeId & peerId)
 {
-    ReleaseSession(FindExistingSession(peerId));
+    ReleaseSession(FindExistingSessionSetup(peerId));
 }
 
 void CASESessionManager::ReleaseSessionsForFabric(FabricIndex fabricIndex)
 {
-    mConfig.devicePool->ReleaseDevicesForFabric(fabricIndex);
+    mConfig.sessionSetupPool->ReleaseDevicesForFabric(fabricIndex);
 }
 
 void CASESessionManager::ReleaseAllSessions()
 {
-    mConfig.devicePool->ReleaseAllDevices();
+    mConfig.sessionSetupPool->ReleaseAllDevices();
 }
 
 CHIP_ERROR CASESessionManager::GetPeerAddress(const ScopedNodeId & peerId, Transport::PeerAddress & addr)
 {
-    OperationalDeviceProxy * session = FindExistingSession(peerId);
-    VerifyOrReturnError(session != nullptr, CHIP_ERROR_NOT_CONNECTED);
-    addr = session->GetPeerAddress();
+    ReturnErrorOnFailure(mConfig.sessionInitParams.Validate());
+    auto optionalSessionHandle = FindExistingSession(peerId);
+    ReturnErrorCodeIf(!optionalSessionHandle.HasValue(), CHIP_ERROR_NOT_CONNECTED);
+    addr = optionalSessionHandle.Value()->AsSecureSession()->GetPeerAddress();
     return CHIP_NO_ERROR;
 }
 
-OperationalDeviceProxy * CASESessionManager::FindExistingSession(const ScopedNodeId & peerId) const
+OperationalSessionSetup * CASESessionManager::FindExistingSessionSetup(const ScopedNodeId & peerId) const
+{
+    return mConfig.sessionSetupPool->FindDevice(peerId);
+}
+
+bool CASESessionManager::DisconnectSession(const ScopedNodeId & peerId) const
```

Comment:
This is searching for existing sessions. `FindExistingSessionSetup` looks for sessions that are in the process of establishing `FindExistingSession`, which is what is used below is going to the sessions manager to fines SessionHandles and mark them as Defunct.

Overall I have this as a helper function, but if you want to remove it I think I can but will need to double check that all current callers to this have access to the SessiongManager to search for the session themselves

Reply:
I stand corrected. Nonetheless, as we discussed, we should remove this method since it's not really in the purview of `CASESessionManager` to handle since it's scope and responsibility is about managing the process of setting up a CASE session. Established sessions are managed by `SessionManager`, so callers should go there directly.